buildpacks/jvm-function-invoker/src/detect.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::Path;
+
+use libcnb::data::build_plan::BuildPlanBuilder;
+use libcnb::{read_toml_file, DetectContext, DetectOutcome, Error, GenericPlatform};
+
+use crate::error::JvmFunctionInvokerBuildpackError;
+use crate::JvmFunctionInvokerBuildpackMetadata;
+use libherokubuildpack::toml_select_value;
+use toml::Value;
+
+pub fn detect(
+    context: DetectContext<GenericPlatform, JvmFunctionInvokerBuildpackMetadata>,
+) -> Result<DetectOutcome, Error<JvmFunctionInvokerBuildpackError>> {
+    let function_toml_path = context.app_dir.join("function.toml");
+    let project_toml_path = context.app_dir.join("project.toml");
+
+    if function_toml_path.exists() || project_toml_salesforce_type_is_function(&project_toml_path) {
+        Ok(DetectOutcome::Pass(
+            BuildPlanBuilder::new()
```

Comment:
https://github.com/Malax/libcnb.rs/blob/0dd2898d23a37911df93452d188c8d3b434ae954/src/data/build_plan.rs#L37-L85

I like this API. Needs some docs.